File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : GenericParams
    {
        public string CurrentUsername { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 120;
        public string Gender { get; set; }
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/user-params.ts
import { GenericParams } from "./generic-params";
import { User } from "./user";

export class UserParams extends GenericParams{
    minAge: number = 18;
    maxAge: number = 120;
    gender: string;
    currentUsername: string;
    orderBy: string = "lastActive";

    constructor(user: User){
        //Gender by default is the opposite
        super();
        this.gender = (user.gender == "male" ? "female" : "male");
        this.currentUsername = user.userName;
    }
}
